pkg/kubelet/kubelet_pods.go

```diff
@@ -1137,6 +1181,12 @@ func (kl *Kubelet) podKiller() {
 				lock.Lock()
 				killing.Delete(string(runningPod.ID))
 				lock.Unlock()
+				kl.podKillingLock.Lock()
+				if gracePeriod, ok := kl.mirrorPodTerminationMap[string(runningPod.ID)]; ok {
```

Comment:
this can add non-mirror pods to the `mirrorPodTerminationMap` right?

Reply:
Please note that we check the presence of UID key in the map.
If the key doesn't exist, there is no entry added.

The removal of the map entry is solely done by kl.deleteOrphanedMirrorPods()